kythe/java/com/google/devtools/kythe/extractors/java/UsageAsInputReportingFileManager.java

```diff
@@ -145,38 +146,87 @@ public void setLocation(Location location, Iterable<? extends File> path) throws
     return fileManager.getLocation(location);
   }
 
-  @Override
+  // TODO(shahms): @Override; added in JDK9
   public Location getLocationForModule(Location location, JavaFileObject fo) throws IOException {
-    return fileManager.getLocationForModule(location, unwrap(fo));
+    // TODO(shahms): return fileManager.getLocationForModule(location, unwrap(fo));
+    try {
+      return (Location)
+          StandardJavaFileManager.class
+              .getMethod("getLocationForModule", Location.class, JavaFileObject.class)
+              .invoke(fileManager, location, unwrap(fo));
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+      throw new IllegalStateException("getLocationForModule called by unsupported Java version", e);
+    }
   }
 
-  @Override
+  // TODO(shahms): @Override; added in JDK9
   public boolean contains(Location location, FileObject fo) throws IOException {
-    return fileManager.contains(location, unwrap(fo));
+    // TODO(shahms): return fileManager.contains(location, unwrap(fo));
+    try {
+      return (Boolean)
+          StandardJavaFileManager.class
+              .getMethod("contains", Location.class, FileObject.class)
+              .invoke(fileManager, location, unwrap(fo));
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+      throw new IllegalStateException("contains called by unsupported Java version", e);
+    }
   }
 
-  @Override
+  // TODO(shahms): @Override; added in JDK9
   public Iterable<? extends JavaFileObject> getJavaFileObjectsFromPaths(
       Iterable<? extends Path> paths) {
-    return Iterables.transform(
-        fileManager.getJavaFileObjectsFromPaths(paths), input -> map(input, null));
+    // TODO(shahms): return Iterables.transform(
+    //      fileManager.getJavaFileObjectsFromPaths(paths), input -> map(input, null));
+    try {
+      return Iterables.transform(
+          (Iterable<? extends JavaFileObject>)
+              StandardJavaFileManager.class
+                  .getMethod("getJavaFileObjectsFromPaths", Iterable.class)
+                  .invoke(fileManager, paths),
+          input -> map(input, null));
+    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+      throw new IllegalStateException(
+          "getJavaFileObjectsFromPaths called by unsupported Java version", e);
+    }
```

Comment:
Generally we'd want to capture the method once in a static, to restrict the reflection (other than the "invoke()") only to startup.

Reply:
That's easy enough to do